pkg/network/vmispec/interface.go

```diff
@@ -58,6 +58,24 @@ func IsPodNetworkWithMasqueradeBindingInterface(networks []v1.Network, ifaces []
 	return true
 }
 
+func ShiftPrimaryInterface(statusIfaces []v1.VirtualMachineInstanceNetworkInterface, networks []v1.Network) []v1.VirtualMachineInstanceNetworkInterface {
+	podNetwork := lookupPodNetwork(networks)
+	if podNetwork == nil {
+		return statusIfaces
+	}
+	if len(statusIfaces) > 0 && statusIfaces[0].Name == podNetwork.Name {
+		return statusIfaces
+	}
+	for index, currStatusIface := range statusIfaces {
+		if currStatusIface.Name == podNetwork.Name {
+			prefix := append([]v1.VirtualMachineInstanceNetworkInterface{statusIfaces[index]},
+				statusIfaces[:index]...)
+			return append(prefix, statusIfaces[index+1:]...)
```

Comment:
Does it still works if index is the last element ? (we have index + 1 here, isn't it out of limits ?)

Reply:
Slice arr[length:] works and returns empty value in contrast to arr[length] which would fail on OutOfRange.
I have a test covering last element use case.